woodwork/column_accessor.py

```diff
@@ -357,6 +361,62 @@ def set_semantic_tags(self, semantic_tags):
             _raise_init_error()
         self._schema._set_semantic_tags(semantic_tags)
 
+    def outliers_dict(self, low_bound=None, high_bound=None):
+        """Gets the values of a series that lay above and below specified bounds.
+
+        Args:
+            series (Series): Data for which the outliers should be determined.
+            low_bound (float, optional): The number below which outliers lay. Is inclusive.
+            high_bound (float, optional): The number above which outliers lay. Is inclusive.
+
+        Note:
+            If neither or only one of low_bound or high_bound is passed in, the bounds will be calculated
+            using the IQR method.
+
+        Returns:
+            dict[str -> float,list[number]]: a dictionary containing outlier values and their corresponding indexes.
+                The following elements will be found in the dictionary:
+
+                - low_values (list[float, int]): the values of the lower outliers
+                - high_values (list[float, int]): the values of the upper outliers
+                - low_indices (list[int]): the corresponding index values for each of the lower outliers
+                - high_indices (list[int]): the corresponding index values for each of the upper outliers
```

Comment:
Maybe there is a reason for this outside of Woodwork, but would it be better to return the results in a dictionaries with the index as the key and the value as, well, the value, instead of separate lists?

Reply:
I now see @davesque also suggested alternate return types below. I don't have any strong opinions either way, just highlighting for discussion.

Reply:
Currently the outlier values aren't sorted in any particular order, but it's very likely that we will want them returned from this function sorted by value, in which case, having them in a list (or series) will be useful